### PR TITLE
Add support for emojis

### LINKED ISSUE #175: Match github emoji support
Github supports a lot of emojis: https://gist.github.com/rxaviers/7360908

Original request by @pauldemarco at https://github.com/dart-lang/pub-dartlang-dart/issues/326.

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #175 :tada:

/cc @isoos

### CODE REVIEW

File: tool/update_emojis.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import 'dart:convert';
+import 'dart:io';
+
+String _emojisJsonRawUrl =
```
Comment (kevmoo) [inline tool/update_emojis.dart]: both of these should be final

**Comment by kevmoo** [comment]
Also look at travis 😄

File: tool/update_emojis.dart (hunk 1 above)
Comment (srawlins) [reply tool/update_emojis.dart]: Done!

File: lib/src/emojis.dart [hunk 2]
```diff
@@ -0,0 +1,1504 @@
+// GENERATED FILE. DO NOT EDIT.
+//
+// This file was generated from emojilib's emoji data file:
+// https://github.com/muan/emojilib/raw/master/emojis.json
+// at 2017-12-28 07:29:53.479234 by the script, tool/update_emojis.dart.
+
+var emojis = <String, String>{
```
[the remark this answers is not in the document]
Comment (srawlins) [reply lib/src/emojis.dart]: Done!

**Comment by srawlins** [comment]
Thank goodness for Travis!

**Pull request merged**